packages/web-sdk/src/instrumentations/session/instrumentation.ts

```diff
@@ -103,17 +91,42 @@ export class SessionInstrumentation extends BaseInstrumentation {
     const sessionTracking = this.config.sessionTracking;
 
     if (sessionTracking?.enabled) {
+      const SessionManager = this.config.sessionTracking?.persistent
+        ? PersistentSessionsManager
+        : VolatileSessionsManager;
+
       const initialSessionMeta = this.createInitialSessionMeta(sessionTracking);
+
+      SessionManager.storeUserSession(
+        createUserSessionObject({
+          sessionId: initialSessionMeta.id,
+          isSampled: Boolean(initialSessionMeta.attributes!['isSampled']),
+        })
+      );
+
+      this.notifiedSession = initialSessionMeta;
       this.api.setSession(initialSessionMeta);
 
-      const sessionManager = this.getSessionManagerInstanceByConfiguredStrategy(this.metas.value.session?.id);
+      const { updateSession } = new SessionManager();
+
+      this.transports?.addBeforeSendHooks(...this.transports.getBeforeSendHooks(), (item) => {
+        updateSession();
+
+        const attributes = item.meta.session?.attributes;
+
+        if (Boolean(attributes?.['isSampled'])) {
+          const { isSampled: _, ...restAttributes } = attributes as any;
```

Comment:
`as any` 🫠

Reply:
Yeah you're rigth it's smelly as hell. 💀
Let me change it.